clickhouse_sqlalchemy/drivers/base.py

```diff
@@ -456,7 +478,27 @@ def _get_col_type(self, name, spec):
         elif spec.startswith('Nullable'):
             inner = spec[9:-1]
             coltype = self.ischema_names['_nullable']
-            return coltype(self._get_col_type(name, inner))
+            return coltype(self._get_column_type(name, inner))
+
+        elif spec.startswith('Enum'):
+            pos = spec.find('(')
+            type = spec[:pos]
+            coltype = self.ischema_names[type]
+
+            options = dict()
+            if pos >= 0:
+                options = self._parse_enum_options(
+                    spec[pos + 1: spec.rfind(')')]
+                )
+            if not options:
+                return sqltypes.NullType
+
+            type_enum = enum.Enum('%s_enum' % name, options)
+
+            def partial_type():
+                return coltype(type_enum)
+
+            return partial_type
```

Comment:
Instead of
```python
            def partial_type():
                return coltype(type_enum)

            return partial_type
```
you can write
```python
            return lambda: coltype(type_enum)
```
This is a little bit more elegant.

Reply:
Good point, change that now.